r-package/grf/R/merge_forests.R

```diff
@@ -32,7 +35,9 @@ merge_forests <- function(forest_list, compute.oob.predictions = TRUE) {
   validate_forest_list(forest_list)
   first_forest <- forest_list[[1]]
 
-  big_forest <- merge(forest_list)
+  big_forest <- merge(lapply(forest_list, get_xptr))
+  big_forest[["_forest"]] <- ForestPtr$new(xptr = big_forest[["_forest"]])
+  lapply(forest_list, clear_xptr) # Invalidate previous forests.
```

Comment:
Why did we start to invalidate the old forests?

Reply:
Because merging is a destructive operation: https://github.com/grf-labs/grf/blob/master/core/src/forest/Forest.h#L52 (which was not an issue before when a copy of the forest was merged)

Reply:
We caught up offline and decided that we're okay to change merging to a destructive operation. We'll rename the method to something like `merge_and_destroy` to be very clear about the behavior.